toolchain/build-toolchain.sh

```diff
@@ -42,15 +45,28 @@ if [ ! -f "$SCRIPT_DIR/../../phoenix-rtos-kernel/Makefile" ] || [ ! -f "$SCRIPT_
     exit 1
 fi
 
+# Those env variables override command line options passed to configure scripts
+# This check was added due to libstdc++ configure using host compiler instead of cross compiler
+# TODO: check if all stages are affected. If not then consider using unset in those stages
+if [[ -v CC || -v CFLAGS || -v LIBS || -v CPPFLAGS || -v CXX || -v CXXFLAGS || -v CPP || -v CXXCPP || -v CXXFILT ]]; then
+  echo "Environment contains variables that should not be set. Abort."
```

Comment:
nitpicking: 4 spaces are used for indentation in other cases :P

Reply:
fixed